### ISSUE TITLE
Doesn't compile with pacman 3.4.0

### ISSUE THREAD

**Issue by OttoAllmendinger** [Kiwi/clyde#61]
Clyde stopped working with pacman-3.4.0. Makepkg fails with this message:

<pre>
gcc -Wall -W -O2 -fPIC  -lalpm `pkg-config --cflags lua` -shared -o lualpm.so lualpm.c -pedantic -D_FILE_OFFSET_BITS=64 -std=c99 -D_GNU_SOURCE
gcc -Wall -W -O2 -fPIC  `pkg-config --cflags lua` -shared -o clydelib/utilcore.so clydelib/utilcore.c -pedantic -D_FILE_OFFSET_BITS=64 -std=c99 -D_GNU_SOURCE
lualpm.c:277:1: error: ‘PM_TRANS_TYPE_UPGRADE’ undeclared here (not in a function)
lualpm.c:277:5: warning: missing initializer
lualpm.c:277:5: warning: (near initialization for ‘transtype_constants[0].value’)
lualpm.c:278:1: error: ‘PM_TRANS_TYPE_REMOVE’ undeclared here (not in a function)
lualpm.c:278:5: warning: missing initializer
lualpm.c:278:5: warning: (near initialization for ‘transtype_constants[1].value’)
lualpm.c:279:1: error: ‘PM_TRANS_TYPE_REMOVEUPGRADE’ undeclared here (not in a function)
lualpm.c:279:5: warning: missing initializer
lualpm.c:279:5: warning: (near initialization for ‘transtype_constants[2].value’)
lualpm.c:280:1: error: ‘PM_TRANS_TYPE_SYNC’ undeclared here (not in a function)
lualpm.c:280:5: warning: missing initializer
lualpm.c:280:5: warning: (near initialization for ‘transtype_constants[3].value’)
lualpm.c:285:1: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘check_transtype’
lualpm.c:292:30: error: expected declaration specifiers or ‘...’ before ‘pmtranstype_t’
lualpm.c: In function ‘push_transtype’:
lualpm.c:294:29: error: ‘t’ undeclared (first use in this function)
lualpm.c:294:29: note: each undeclared identifier is reported only once for each function it appears in
lualpm.c:294:5: warning: passing argument 2 of ‘push_constant’ makes integer from pointer without a cast
lualpm.c:222:1: note: expected ‘int’ but argument is of type ‘const struct constant_t *’
lualpm.c: In function ‘lalpm_pkg_get_removes’:
lualpm.c:806:5: warning: implicit declaration of function ‘alpm_pkg_get_removes’
lualpm.c:806:25: warning: initialization makes pointer from integer without a cast
lualpm.c: In function ‘lalpm_trans_get_type’:
lualpm.c:1910:5: error: ‘pmtranstype_t’ undeclared (first use in this function)
lualpm.c:1910:5: warning: statement with no effect
lualpm.c:1910:19: error: expected ‘;’ before ‘x’
lualpm.c:1911:9: error: ‘x’ undeclared (first use in this function)
lualpm.c:1915:5: error: too many arguments to function ‘push_transtype’
lualpm.c:292:1: note: declared here
lualpm.c: In function ‘lalpm_trans_get_pkgs’:
lualpm.c:1932:5: warning: implicit declaration of function ‘alpm_trans_get_pkgs’
lualpm.c:1932:25: warning: initialization makes pointer from integer without a cast
lualpm.c: In function ‘lalpm_trans_init’:
lualpm.c:2211:5: error: ‘pmtranstype_t’ undeclared (first use in this function)
lualpm.c:2211:5: warning: statement with no effect
lualpm.c:2211:19: error: expected ‘;’ before ‘type’
lualpm.c:2218:40: error: ‘type’ undeclared (first use in this function)
lualpm.c:2221:40: error: incompatible type for argument 1 of ‘alpm_trans_init’
/usr/include/alpm.h:405:5: note: expected ‘pmtransflag_t’ but argument is of type ‘const struct constant_t *’
lualpm.c:2221:40: error: incompatible type for argument 2 of ‘alpm_trans_init’
/usr/include/alpm.h:405:5: note: expected ‘alpm_trans_cb_event’ but argument is of type ‘pmtransflag_t’
lualpm.c:2221:40: warning: passing argument 3 of ‘alpm_trans_init’ from incompatible pointer type
/usr/include/alpm.h:405:5: note: expected ‘alpm_trans_cb_conv’ but argument is of type ‘void (*)(enum pmtransevt_t,  void *, void *)’
lualpm.c:2221:40: warning: passing argument 4 of ‘alpm_trans_init’ from incompatible pointer type
/usr/include/alpm.h:405:5: note: expected ‘alpm_trans_cb_progress’ but argument is of type ‘void (*)(enum pmtransconv_t,  void *, void *, void *, int *)’
lualpm.c:2221:40: error: too many arguments to function ‘alpm_trans_init’
/usr/include/alpm.h:405:5: note: declared here
lualpm.c: In function ‘lalpm_trans_sysupgrade’:
lualpm.c:2230:5: warning: implicit declaration of function ‘alpm_trans_sysupgrade’
lualpm.c: In function ‘lalpm_trans_addtarget’:
lualpm.c:2242:5: warning: implicit declaration of function ‘alpm_trans_addtarget’
make: *** [lualpm] Error 1
make: *** Waiting for unfinished jobs....
gcc -Wall -W -O2 -fPIC  -shared -o clydelib/signal.so clydelib/lsignal.c
clydelib/lsignal.c: In function ‘sighook’:
clydelib/lsignal.c:167:46: warning: unused parameter ‘ar’
clydelib/lsignal.c: In function ‘l_signal’:
clydelib/lsignal.c:231:10: warning: ‘sig’ may be used uninitialized in this function
</pre>

**Comment by juster** [comment]
This is a duplicate of issue #58 (which I am now closing)

However, this is much better error reporting =) thanks.  I like this one better.  I will leave this issue open for anyone else having the same problem... until the official repo is updated.

I think you have already found my fork (thanks for the patch!).  We will soon be merging my forked copy of clyde into the official copy.  Until then you can build my fork like so:

  cd /tmp
  wget xrl.us/clydepkg
  makepkg
  pacman -U ...

The built package is called _clyde-fork-git_ in order to differentiate it from the official _clyde-git_ package.

**Comment by juster** [comment]
The fork has been merged to Kiwi's master branch.  This issue is solved, closing.
